src/env.cr

```diff
@@ -37,14 +39,22 @@ module ENV
     end
   end
 
+  def self.clear
+    keys.each { |k| delete k }
+  end
+
   def self.each
     environ_ptr = LibC.environ
     while environ_ptr
       environ_value = environ_ptr.value
       if environ_value
         key_value = String.new(environ_value)
-        key, value = key_value.split '=', 2
-        yield key, value
+        kv = key_value.split '=', 2
+        key = kv[0]
+        val = kv[0]? || ""
+        yield({key, val})
+# one day...
+#        yield(key, val)
```

Comment:
I never understood the point of Hash being Enumerable in Ruby. If you do `select`, `reject`, etc., you get arrays, and you probably wanted a Hash as a result, so you end up doing `Hash[hash.select { ... }]`, which is long to type and slow.

I think in this case I'd prefer to have specialized Hash methods that return hashes in these cases. One exception is `each_with_index`, but we already implemented that. I don't think `map` makes sense either.

And in the case you do want to treat Hash as an enumerable of tuples, you can do `hash.each`.